### PR TITLE
[iOS] Migrate AppDelegate in bare-expo to Swift

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

Migrating bare-expo to use the AppDelegate in Swift

# How

- Created `AppDelegate.swift` that already inherits from `ExpoAppDelegate`.
- Removed `AppDelegate.h`, `AppDelegate.mm`, `main.m` and `noop-file.swift` files.

# Test Plan

bare-expo compiles, launches and I confirmed that app delegate subscribers' methods are invoked (specifically I was testing `expo-splash-screen` and `expo-screen-orientation`)

### CODE REVIEW

File: apps/bare-expo/ios/BareExpo/AppDelegate.mm [hunk 1]
```diff
@@ -1,62 +0,0 @@
...
-- (NSURL *)bundleURL
-{
-#if DEBUG
-  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@".expo/.virtual-metro-entry"];
-#else
-  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
-#endif
-}
-
-// Linking API
```
Comment (Kudo) [inline apps/bare-expo/ios/BareExpo/AppDelegate.mm:33]: should we keep the following api to keep deep linking and notification works?

Comment (Kudo) [reply apps/bare-expo/ios/BareExpo/AppDelegate.mm:33]: for references, the notification relates api were added by #17201

Comment (tsapeta) [reply apps/bare-expo/ios/BareExpo/AppDelegate.mm:33]: You're right. Also leaving some functions there makes it more clear for users that they should call super methods when they override something 👍

**Review by Kudo — APPROVED** [review]
talking with Tomasz that we are removing notification subscribers and let react-native-firebase use their config-plugin to add the subscribers.

**Pull request merged**